fakestorage/server.go

```diff
@@ -193,6 +193,7 @@ func (s *Server) buildMuxer() {
 		r.Path("/b/{bucketName}/o").Methods(http.MethodGet).HandlerFunc(jsonToHTTPHandler(s.listObjects))
 		r.Path("/b/{bucketName}/o").Methods(http.MethodPost).HandlerFunc(jsonToHTTPHandler(s.insertObject))
 		r.Path("/b/{bucketName}/o/{objectName:.+}").Methods(http.MethodPatch).HandlerFunc(jsonToHTTPHandler(s.patchObject))
+		r.Path("/b/{bucketName}/o/{objectName:.+}").Methods(http.MethodPut, http.MethodPost).HandlerFunc(jsonToHTTPHandler(s.updateObject))
```

Comment:
nevermind just saw the comment here https://github.com/fsouza/fake-gcs-server/pull/434#issuecomment-963984673

Reply:
As you noted the Go SDK does not support the operations, and I don't know enough Go to write the client code to test it :( 

I guess it would be something like this and replace the method objHandle.UpdateXXX with a new client implementation:

```
func TestServerClientObjectUpdateMetadata(t *testing.T) {
	const (
		bucketName  = "some-bucket"
		objectName  = "items/data.txt"
		content     = "some nice content"
		contentType = "text/plain; charset=utf-8"
	)
	objs := []Object{
		{
			ObjectAttrs: ObjectAttrs{
				BucketName:  bucketName,
				Name:        objectName,
				ContentType: contentType,
			},
			Content: []byte(content),
		},
	}
	runServersTest(t, objs, func(t *testing.T, server *Server) {
		client := server.Client()
		objHandle := client.Bucket(bucketName).Object(objectName)

		metadata := map[string]string{
			"1-key": "1-value",
			"2-key": "2-value",
		}
		testUpdate(metadata, metadata, objHandle, t)

		metadata = map[string]string{
			"1-key": "1.1-value",
			"3-key": "3-value",
		}
		testUpdate(metadata, metadata, objHandle, t)
	})
}

func testUpdate(newMetadata, finalMetadata map[string]string, objHandle *storage.ObjectHandle, t *testing.T) {
	ctx := context.TODO()
	_, err := objHandle.UpdateXXX(ctx, storage.ObjectAttrsToUpdate{Metadata: newMetadata})
	if err != nil {
		t.Fatal(err)
	}
	attrs, err := objHandle.Attrs(ctx)
	if err != nil {
		t.Fatal(err)
	}
	checkObjectMetadata(attrs.Metadata, finalMetadata, t)
}
```

Reply:
I'll take care of the test. Thank you for contributing!